File: Aigilas/Aigilas/Strategies/AttackSelfStrategy.cs
using Aigilas.Creatures;
using Aigilas.Skills;

namespace Aigilas.Strategies
{
    public class AttackSelfStrategy : IStrategy
    {
        public AttackSelfStrategy(ICreature parent)
            : base(parent,Strategy.AttackSelf)
        {
        }

        public override void Act()
        {
            if (AbleToMove())
            {
                _parent.ApplyDamage(1);
            }
        }
    }
}

File: proj/src/aigilas/strategies/impl/AttackSelfStrategy.java
package aigilas.strategies.impl;

import aigilas.creatures.BaseCreature;
import aigilas.strategies.BaseStrategy;
import aigilas.strategies.Strategy;

public class AttackSelfStrategy extends BaseStrategy {
    public AttackSelfStrategy(BaseCreature parent)

    {
        super(parent, Strategy.AttackSelf);

    }

    @Override
    public void act() {
        if (AbleToMove()) {
            _parent.applyDamage(1);
        }
    }
}
